superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -287,6 +344,175 @@ export default function DataSourcePanel({
     [lists.columns, showAllColumns],
   );
 
+  const handleOverwriteDataset = async () => {
```

Comment:
can most of this functionality re saving/overwriting the dataset be moved into the dataset modal instead?